csrc/device_lower/analysis/tensor_memory.cpp

```diff
@@ -193,7 +198,38 @@ TMemAlllocationInfo computeTMemAlllocationInfo(Fusion* fusion) {
 std::pair<
     std::unordered_map<TensorView*, TMemRegisterDataPath>,
     std::unordered_map<TensorView*, TMemRegisterDataPath>>
-computeTMemLdStDataPath(Fusion* fusion) {
+computeTMemLdStDataPath(Fusion* fusion, const TMemAlllocationInfo& allocation) {
```

Comment:
I'd appreciate if you could break up this function into more semantically closed smaller groups. For example, I think lines 264-321 can be easily form a single separate function like `getContiguityOfParallelTypes`. I believe that would make this function easier to grasp.

Reply:
I extracted out `getInterestedThreadParallelTypes` and `getThreadParallelTypesMergedByContiguity`.